readme.md

```diff
@@ -1,48 +1,408 @@
+  - [Creating a Presentation From Scratch](#creating-a-presentation-from-scratch)
+  - [Modifying an existing Presentation](#modifying-an-existing-presentation)
+  - [Saving A Presentation](#saving-a-presentation)
+  - [Setting Properties](#setting-properties)
+  - [Setting Layouts](#setting-layouts)
+  - [Setting Text Direction](#setting-text-direction)
+- [Slides](#slides)
+  - [Adding Slides](#adding-slides)
+  - [Removing Slides](#removing-slides)
+  - [Reordering Slides](#reordering-slides)
+  - [Formatting Options](#formatting-options)
+  - [Applying Master Slides](#applying-master-slides)
+  - [Adding Slide Numbers](#adding-slide-numbers)
+- [Adding Content to Slides](#adding-content-to-slides)
+  - [Charts](#charts)
+    - [Bar Charts](#bar-charts)
+  - [Images](#images)
+  - [Media Objects](#media-objects)
+  - [Shapes](#shapes)
+  - [Text Boxes](#text-boxes)
+- [Testing](#testing)
+- [Contributing](#contributing)
+- [Special Thanks](#special-thanks)
+- [License](#license)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Getting Started
+
+```bash
 $ npm install node-pptx
 ```
 
-## Usage
+Let's create a very simple presentation with one slide.
 
 ```javascript
-const pptx = require('node-pptx');
+const PPTX = require('node-pptx');
+let pptx = new PPTX.Composer();
 
-// In the futures presentations will also be able to composed through the DSL
-new pptx.Presentation()
-  .compose(pres => {
+await pptx.compose(pres => {
+    pres.addSlide(slide => {
+        slide
+            .addText(text => {
+                text.value('Hello World');
+            });
+    });
+});
 
-    pres.title('My Presentation')
-    .addSlide(slide => {
+await pptx.save(`./hello-world.pptx`);
+```
+## General Conventions
+`node-pptx` has a friendly declarative DSL to quickly design a pptx file. This makes your JavaScript code very readable because, it allows you to visually segment and compartmentalize your code to the presentation element you are trying to edit. Here is a simple example of adding a text box to a slide:
 
-      slide
-      .addText(text => {
+```javascript
+await pres.addSlide(slide => {
 
+    // declarative way of adding an object
+    slide.addText(text => {
         text
-        .value('Hello world')
-        .x(10)
-        .y(0);
-      })
-      .addShape(shape => {
+            .value('Hello World!')
+            .x(100)
+            .y(50)
+            .fontFace('Alien Encounters')
+            .fontSize(20)
+            .textColor('CC0000')
+            .textWrap('none')
+            .textAlign('left')
+            .textVerticalAlign('center')
+            .line({ color: '0000FF', dashType: 'dash', width: 1.0 })
+            .margin(0);
+    });
+});
+```
+You can also achieve the same result using the more terse object-only format by supplying a configuration object instead of a function.
+
+*Note*: Not all presentation elements support the object-only format.
```

Comment:
Yes, they do.

Reply:
For example, `Slide` doesn't right?

Reply:
I meant the objects which sit on slides (text boxes, shapes, etc.).